src/library_manager/lib_manager.c

```diff
@@ -641,7 +641,7 @@ int lib_manager_register_module(const uint32_t component_id)
 	}
 
 	mod = (struct sof_man_module *)((const uint8_t *)desc + SOF_MAN_MODULE_OFFSET(entry_index));
-	const struct sof_uuid *uid = (struct sof_uuid *)&mod->uuid[0];
+	const struct sof_uuid *uid = (struct sof_uuid *)&mod->uuid;
```

Comment:
you could move those two hunks to this commit as "preparation" or move all three changes to where you change to the structure